src/transformStateWithClones.js

```diff
@@ -7,7 +7,60 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let clones = [state];
+
+  actions.forEach(handleAction);
+
+  clones.shift();
+
+  return clones;
+
+  function handleAction({ type, extraData, keysToRemove }) {
+    const lastState = clones[clones.length - 1];
```

Comment:
If you want to use helping functions, make them pure. 
The pure function should take some parameters and return the result, which you should use later. So I think there is no reason to use the helping function in this task.